docs/reference/aggregations/metrics/geocentroid-aggregation.asciidoc

```diff
@@ -144,6 +144,86 @@ The response for the above aggregation:
 --------------------------------------------------
 // TESTRESPONSE[s/\.\.\./"took": $body.took,"_shards": $body._shards,"hits":$body.hits,"timed_out":false,/]
 
+
+[discrete]
+[role="xpack"]
+==== Geo Centroid Aggregation on `geo_shape` fields
+
+The centroid metric for geo-shapes is more nuanced than for points. The centroid of a specific aggregation bucket
+containing shapes is the centroid of the highest-dimensionality shape type in the bucket. For example, if a bucket contains
+shapes comprising of polygons and lines, then the lines do not contribute to the centroid metric. Each type of shape's
+centroid is calculated differently.
+
+|===
+|Geometry Type | Centroid Calculation
+
+|[Multi]Point
+|equally weighted average of all the coordinates
+
+|[Multi]LineString
+|a weighted average of all the centroids of each segment, where the weight of each segment is its length in degrees
+
+|[Multi]Polygon
+|a weighted average of all the centroids of all the triangles of a polygon where the triangles are formed by every two consecutive vertices and the starting-point.
+ holes have negative weights. weights represent the area of the triangle in deg^2 calculated
+
+|GeometryCollection
+|The centroid of all the underlying geometries with the highest dimension. If Polygons and Lines and/or Points, then lines and/or points are ignored.
+ If Lines and Points, then points are ignored
+|===
```

Comment:
Not very important but should be add rectangle / envelope?

Reply:
good question. I did not consider it because I was not sure how anyone would create them? We do not support rectangle in the new indexing strategy, right?

Reply:
Yes, it is just rated as a 4 points polygon

Reply:
Maybe we can just add a note that envelopes and circles (ingested through the processor) are treated as polygons?

Reply:
that is a good idea. will add that in! thanks